### ISSUE TITLE
A11y improvements

### ISSUE THREAD

**Issue by remischwartz** [Sqrrl/wc-datepicker#25]
Hi!

Thank you for your amazing library! We used it to build our accessible web components design system.

We received some feedback from our a11y consultant, and he identified some improvement opportunities:

- Add an `aria-label` attribute to the `th` elements. The `abbr` attribute is not announced by all screen readers.
- Only change the value of the hidden span in `wc-datepicker__header` when the selected value changes (currently, it changes when users navigate through the dates with the keyboard).
- In range mode, change the value of the hidden span in `wc-datepicker__header` to express the selected range (e.g., “June 21, 2024 to June 25, 2024”).
- Add a `title` attribute to the month and year selection fields (same value as the aria-label).
- When using `disableDate`, prevent the disabled dates from being focused.

Do you think these changes could be implemented?
I can open PRs if you prefer.

**Comment by Sqrrl** [comment]
Hey. Thanks for your feedback. Great insight. I would like to incorporate it, just don‘t know when I‘ll get to it. A PR would be great.

**Comment by remischwartz** [comment]
Hi Philipp,
I created a PR to covers the comments except for the focus on disabled dates. 
I am still exploring what would be the best behavior in this case, I will create a separate PR when I have a satisfying solution.
